index-handlers/src/main/java/no/sikt/nva/nvi/index/query/QueryFilterType.java

```diff
@@ -0,0 +1,42 @@
+package no.sikt.nva.nvi.index.query;
+
+import static java.util.Objects.isNull;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum QueryFilterType {
+  COLLABORATION("collaboration"),
+  OTHERS_APPROVE("approvedByOthers"),
+  OTHERS_REJECT("rejectedByOthers"),
+  NEW_AGG("pending"),
+  NEW_COLLABORATION_AGG("pendingCollaboration"),
+  PENDING_AGG("assigned"),
+  PENDING_COLLABORATION_AGG("assignedCollaboration"),
+  APPROVED_AGG("approved"),
+  APPROVED_COLLABORATION_AGG("approvedCollaboration"),
+  REJECTED_AGG("rejected"),
+  REJECTED_COLLABORATION_AGG("rejectedCollaboration"),
+  DISPUTED_AGG("dispute"),
+  ASSIGNMENTS_AGG("assignments"),
+  EMPTY_FILTER("");
+
+  private final String filter;
+
+  QueryFilterType(String filter) {
+    this.filter = filter;
+  }
+
+  public static Optional<QueryFilterType> parse(String candidate) {
```

Comment:
The use of `candidate` here, while correct linguistically, conflicts with the domain term "candidate"

Reply:
Agreed. This is the original code though, I just moved it as-is. I can rename it now for clarity since I'm at it.